### DISCUSSION TITLE
Reading published OpenDDSharp topics in third party DDS topic reader

### DISCUSSION THREAD

**Discussion by mellegge** [jmmorato/openddsharp#168] [Q&A] [answered]
I just created the getting started apps (Publisher and Subscriber).  The one topic gets sent from the publisher and is read by the subscriber as expected.  For interop purposes I was hoping to be able to read the topic on a third party DDS reader (from RTI).  Not having much luck yet.

For one thing, the publisher writes the topic once and it is immediately consumed by the subscriber.  I changed the publisher to write the topic every 500msec and tried both with and without a subscriber, but the third party reader does not see it.

Maybe the issue is QoS.  It tried to change the persistence of the topic but I have not had much luck - not sure of the right approach.

Note I changed the third party reader to look at Domain ID 42 as per the getting started apps.

Anyone have any suggestions to help?

Thanks
Mel

**Comment by jmmorato** [comment]
It could be multiple things, transport configuration, CDR serialisation used, QoS compatibility...

Activate logging in the factory
`var dpf = ParticipantService.Instance.GetDomainParticipantFactory("-DCPSConfigFile", "rtps.ini", "-DCPSDebugLevel", "10", "-ORBLogFile", "LogFile.log", "-ORBDebugLevel", "10");`

and also in the RTI side with
```xml
<participant_factory_qos>
        <logging>
          <verbosity>ALL</verbosity>
          <print_format>DEFAULT</print_format>
          <output_file>E:\DDSLog.txt</output_file>
        </logging>
 </participant_factory_qos>
```

and let's take a look inside

**Comment by jmmorato** [comment]
There is (was) and option in older OpenDDS version to ensure the serialization compatibility `-DCPSRTISerialization 1` not sure if still required as OpenDDS is using CDR2 as default now, but you can give it a try

**Reply by jmmorato** [reply]
As per this comment https://github.com/OpenDDS/OpenDDS/issues/1813#issuecomment-669196694 look it is activated by default after OpenDDS v3.14...

**Comment by jmmorato** [comment]
Attached a basic reproducer to check interoperatibility
[Interoperatibility.zip](https://github.com/jmmorato/openddsharp/files/11400545/Interoperatibility.zip)

Required XCDR2 configuration on RTI app to make it work with the default QoS.

**Comment by mellegge** [comment, marked as answer]
There were two problems:
1) The RTI third party tool needed the XCDR2 representation set (in my case in the QoS file)
2) The RTI third party tool was using the wrong multicast address for discovery.  My NDDS_DISCOVERY_PEERS environment variable was set to something other than the default of 239.255.0.1
